runtime/dart_isolate.h

```diff
@@ -125,6 +127,9 @@ class DartIsolate : public UIDartState {
   std::vector<std::shared_ptr<const fml::Mapping>> kernel_buffers_;
   std::vector<std::unique_ptr<AutoFireClosure>> shutdown_callbacks_;
   ChildIsolatePreparer child_isolate_preparer_;
+  bool terminate_child_isolates_;
+  // TODO(garyq): Prevent memoryleaks by garbage collecting/parent callback
+  std::vector<std::weak_ptr<DartIsolate>> child_isolates_;
```

Comment:
I believe the leak is just the weak pinter to the isolate object and not the object itself. You can store a backpointer to the parent for every isolate that is created.